include/ttmlir/Dialect/TTIR/IR/TTIROps.td

```diff
@@ -340,6 +340,62 @@ def TTIR_TransposeOp : TTIR_DPSOp<"transpose"> {
     let hasVerifier = 1;
 }
 
+def TTIR_Conv2dOp : TTIR_DPSOp<"conv2d"> {
+    let summary = "Conv2d operation.";
+    let description = [{
+      Conv2d operation.
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         AnyRankedTensor:$weight,
+                         AnyRankedTensor:$bias,
+                         AnyRankedTensor:$output,
+                         SI32Attr:$kernel_height,
+                         SI32Attr:$kernel_width,
+                         SI32Attr:$stride_height,
+                         SI32Attr:$stride_width,
+                         SI32Attr:$dilation,
+                         SI32Attr:$groups,
+                         SI32Attr:$padding_left,
+                         SI32Attr:$padding_right,
+                         SI32Attr:$padding_top,
+                         SI32Attr:$padding_bottom,
+                         SI32Attr:$is_convtranspose2d,
+                         SI32Attr:$output_height_transpose,
+                         SI32Attr:$output_width_transpose,
+                         SI32Attr:$stride_transpose,
+                         TT_OperandConstraintArrayAttr:$operand_constraints);
+
+    let results = (outs AnyRankedTensor:$result);
+
+    let extraClassDeclaration = [{
+      MutableOperandRange getDpsInitsMutable() { return getOutputMutable(); }
+    }];
+
+    let hasVerifier = 1;
+}
+
+def TTIR_ConvertConvWeightTensorToTiledLayoutOp : TTIR_DPSOp<"convert_conv_weight_tensor_to_tiled_layout"> {
```

Comment:
Is this more of a "pass" rather than an "op"?

Reply:
Yes, we should move this. There's two places I think this might go:
1. `TTIRLayout` pass (`lib/Dialect/TTIR/Transforms/Passes.cpp`)
2. `TTIRToTTNN` dialect conversion (`lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp`)

I think `1` is a lot cleaner, we should do it there.

Reply:
So I added this because it's an actual op in TTNN. What I didn't realize is that its also called within the TTNN Conv2D op, so I actually don't need it to bringup Conv2D.